Changes 'the' to 'then' in stateful goroutine example

ashish91 opened the pull request.

eliben reviewed (approved):
Thanks for the typo fix.

Please run the `build` tool as described in https://github.com/mmcgrana/gobyexample/blob/master/CONTRIBUTING.md to generate the changed HTML as well, and add it to your PR. This way I can merge the PR with your contribution

ashish91 commented:
Hey @eliben, I rebased the branch from master and ran tools/build on my local. Few other changes were also added after running tools/build. Is that ok ?
